task-maker-format/src/ioi/sanity_checks/att.rs

```diff
@@ -400,3 +402,42 @@ impl SanityCheck<IOITask> for AttWithNoCheck {
         Ok(())
     }
 }
+
+/// Check that the template and grader in att compile together
+#[derive(Debug, Default)]
+pub struct AttTemplatesShouldCompile;
+
+impl SanityCheck<IOITask> for AttTemplatesShouldCompile {
+    fn name(&self) -> &'static str {
+        "AttTemplatesShouldCompile"
+    }
+
+    fn pre_hook(&mut self, task: &IOITask, eval: &mut EvaluationData) -> Result<(), Error> {
+        for grader in task.grader_map.all_paths() {
+            let ext = grader
+                .extension()
+                .ok_or_else(|| anyhow!("Grader has no extension"))?
+                .to_string_lossy();
+            let att_name = format!("att/{}.{}", task.name, ext);
+            let template = task.path.join(&att_name);
+
+            let grader_name = grader
+                .file_name()
+                .ok_or_else(|| anyhow!("Grader has no file name"))?
+                .to_string_lossy();
+            let att_grader = format!("att/{}", grader_name);
```

Comment:
Note that this is using only the attached graders, not the official ones. Probably we want to check both the graders (attach template with attached grader and attached template with official grader).

Reply:
We already check the (template; official grader) pair in most (all?) problems since we symlink the template in sol, usually as `template.cpp`. I guess this would allow us to remove that symlink

Reply:
Ah, that's true. I'd keep the symlink but it's not obvious if we want to always check this with a sanity check or just trust the symlink is there.

I've just noticed that `AttTemplates` checks that there is a template file (using the `<taskname>.cpp` format), but it doesn't check that it is a symlink (maybe we should extend that check!).

Reply:
Since we're already checking that a template exists, a "this should be a symlink to ../sol/template.<ext>" warning seems quite logical. I can add it to this PR or another one if you prefer

Reply:
Another PR please.